### DISCUSSION TITLE
419 in Livewire component when it's used on the 404 page template

### DISCUSSION THREAD

**Discussion by smares** [livewire/livewire#4219] [1. Help]
I have a Livewire component that I am using in a base template that defines the general look and feel of the site (e.g. header, footer, menu, content area). Now I also went ahead and designed a custom 404 page that uses this base template. I placed it inside `views/errors/404.blade.php` so it gets picked up by Laravel whenever a 404 occurs. The only problem is that Livewire has no CSRF token and as soon as the 404-page loads, I get an alert that the page expired and needs to be reloaded. `window.livewire_token` is empty.

Am I missing something?

**Comment by joshhanley** [comment]
@smares hmm I think there is something extra you need to do to get Livewire running on the 404 page, but I can't remember exactly what.

I'd be tracking down to see why the CSRF token isn't coming through, maybe it's a layout file issue or something?

**Reply by smares** [reply]
I think this nails it: https://github.com/laravel/framework/issues/11653 `csrf_token()` doesn't return anything because the session didn't start for 404. 😢

**Reply by joshhanley** [reply]
@smares ah yep, knew there was something different about 404 pages, thanks for confirming! I think I either had to start the session for a 404 page (it's been a while), or not use Livewire, I can't remember what I did.

**Reply by smares** [reply]
I solved my problem by adding a [fallback route](https://laravel.com/docs/8.x/routing#fallback-routes) to my routes files that returns the 404 view along with the 404 status code. Everything works as expected now. 😄

**Reply by joshhanley** [reply]
Nice! Good solution 😁
